### ISSUE TITLE
calcite select component - add method to open

### ISSUE THREAD

**Issue by canadajebus** [Esri/calcite-components#5294]
### Description

The only method available on the calcite select component is setFocus().  This just seems to highlight the border of the element.  I'd like to have access to an open() method that opens the select / drop down.  This would be callable when another element such as the div/span beside it is clicked.  

### Acceptance Criteria

See above.

### Which Component

select

### Example Use Case

Basically I want to but some text next to the select that when clicked on also opens the select.

**Comment by jcfranco** [comment]
Thanks for the enhancement request. Unfortunately, the current implementation relies on `<select>`, which does not expose a way for us to support this.

Stepping back a bit, could you break down your use case?

**Comment by canadajebus** [comment]
Thanks for the reply.  Basically the div element that is the parent to the select component I'm creating, I want to wire a click event to.  I want that click event on the larger div to open the select element.

Basically my code so far looks like this:

```
// Create the layer category dropdown
this.controls.btnCategoryFilter.domNode = domConstruct.create("div",{
  id: "allLayersControl_btnCategoryFilter",
  onclick: (event) => {
    if (event.target != this.controls.btnCategoryFilter.select)
    {
      // console.log(event);
      // To Do: Open the dropdown when calcite select component adds open() method
      this.controls.btnCategoryFilter.select.setFocus();
    }
  }
}, this.controlsNode);

this.controls.btnCategoryFilter.label = domConstruct.create("span",{
  id: "allLayersControl_btnCategoryFilterLabel",
  innerHTML: "Filter by Category: "
}, this.controls.btnCategoryFilter.domNode);

this.controls.btnCategoryFilter.select = domConstruct.create("calcite-select",{
  id: "allLayersControl_btnCategoryFilterSelect",
  value: this.layerCategories[0].name,
  calciteSelectChange: (event) => {
    console.log(event);
    console.log(event.target.value);
  }
}, this.controls.btnCategoryFilter.domNode);

for (l in this.layerCategories)
{
  this.controls.btnCategoryFilter.selectItems.push(domConstruct.create("calcite-option",{
    label: this.layerCategories[l].name,
    value: this.layerCategories[l].name
  }, this.controls.btnCategoryFilter.select));
}
```


Screenshot of current display with the parent div of the select highlighted in Chrome:
![image](https://user-images.githubusercontent.com/13382047/189460007-11c36db0-be68-4ee3-9d80-f9ff5db6503a.png)

**Comment by jcfranco** [comment]
Gotcha, thanks for sharing. I'm afraid we can't do much else regarding opening programmatically.

Although, not exactly what you requested, if you use a `calcite-label` as the parent, clicking on it will focus the select and allow users to open it via the keyboard or simply by clicking it. This brings it closer to the native HTML experience.

**Comment by jcfranco** [comment]
Closing based on ☝️.
